scipy/_lib/_util.py

```diff
@@ -815,15 +814,17 @@ def _rng_spawn(rng, n_children):
     return child_rngs
 
 
-def _get_nan(*data):
+def _get_nan(*data, xp=None):
+    xp = array_namespace(*data) if xp is None else xp
     # Get NaN of appropriate dtype for data
-    data = [np.asarray(item) for item in data]
+    data = [xp.asarray(item) for item in data]
     try:
-        dtype = np.result_type(*data, np.half)  # must be a float16 at least
+        min_float = getattr(xp, 'float16', xp.float32)
+        dtype = xp.result_type(*data, min_float)  # must be at least a float
     except DTypePromotionError:
```

Comment:
Do all libraries raise the same exception here? It wasn't clear to me from the array API docs that they do?

Reply:
If the user is passing arrays with dtypes specified by the array API, I don't think this will happen, so I don't think we need to worry about it. They will get an error about the dtypes being wonky and they can use more standard dtypes.